content/en/blog/_posts/2021-XX-XX-writing-an-operator-for-pod-labels.md

```diff
@@ -0,0 +1,447 @@
+---
+layout: blog
+title: "Writing an Operator for Pod Labels"
+date: 2021-03-XX
+slug: writing-an-operator-for-pod-labels
+---
+
+**Authors**: Arthur Busser, Site Reliability Engineer, Padok
+
+[Operators][what-is-an-operator] are proving to be an excellent solution to
+running stateful distributed applications in Kubernetes. Open Source tools like
+the [Operator SDK][operator-sdk] provide ways to build reliable and maintainable
+operators, and extending Kubernetes to implement custom scheduling needs is now
+easier than ever.
+
+This post will walk you through writing a Kubernetes operator in Go. The
+operator will add a `pod-name` label to Pods that have a specific annotation.
+
+## Why do we need an operator for this?
+
+Kubernetes operators run complex software inside your cluster. The open-source
+community has already built [many operators][operatorhub] for distributed
+applications like Prometheus, ElasticSearch, or ArgoCD. Even outside of
+open-source, operators can help to bring new functionality to your Kubernetes
+cluster.
+
+I recently worked on a project where we needed to create a Service that routed
+traffic to a specific Pod in a ReplicaSet. The problem is that a Service can
+only select Pods by label, and all pods in a ReplicaSet have the same labels.
+How could we add a label to these pods with a unique value for each? The answer
+is: with an operator.
+
```

Comment:
You might want to talk about the alternative: create a Service without a selector, and manage the EndpointSlices (or Endpoints) for that Service directly. You would still need to write a controller to make that possible.

Reply:
Done 👍